### DISCUSSION TITLE
Single file templates

### DISCUSSION THREAD

**Discussion by pylang** [copier-org/copier#346] [Q&A] [answered]
Can copier template single files?  From what I recall, [cookiecutter does not handle](https://github.com/cookiecutter/cookiecutter/pull/885) this. 

Rather than having separate files as templates:

```bash
└───expected/
        bar.md
        baz.rst
        foo.txt
```

Cookiecutter requires directories:

```bash
└───actual/
    ├───bar/
    │   │   cookiecutter.json
    │   │
    │   └───{{cookiecutter.project_dir}}
    │           {{cookiecutter.file_name}}.md
    │
    ├───baz/
    │   │   cookiecutter.json
    │   │
    │   └───{{cookiecutter.project_dir}}
    │           {{cookiecutter.file_name}}.rst
    │
    └───foo/
        │   cookiecutter.json
        │
        └───{{cookiecutter.project_dir}}
                {{cookiecutter.file_name}}.txt

```

That's a bit complicated.  How does copier approach this?

**Comment by yajo** [comment, marked as answer]
I recall this appeared before... yes! This might answer your question: https://github.com/copier-org/copier/issues/231#issuecomment-661667708
